src/file.cpp

```diff
@@ -974,20 +968,28 @@ namespace libtorrent
 	{
 #if defined TORRENT_WINDOWS && !defined TORRENT_MINGW
 #if TORRENT_USE_WSTRING
-		wchar_t* cwd { ::_wgetcwd(nullptr, 0) };
-		std::shared_ptr<wchar_t> holder(cwd, &std::free);
-		std::string ret { cwd == nullptr ? "" : convert_from_wstring(cwd) };
+#define GetCurrentDir_ ::_wgetcwd
+#define PathCharType_ wchar_t
```

Comment:
this change made it even more verbose. Also, for type aliases, please use: ``using <name> = <type>;``